docs/intree-migrate.md

```diff
@@ -0,0 +1,148 @@
+
+## RBD
+
+- [Prerequisite](#prerequisite)
+- [Volume operations after enabling CSI migration](#volume-operations-after-enabling-csi-migration)
+  - [Create volume](#create-volume)
+  - [Mount volume to a POD](#mount-volume-to-a-pod)
+  - [Resize volume](#resize-volume)
+  - [Unmount volume](#unmount-volume)
+  - [Delete volume](#delete-volume)
+- [References](#additional-references)
+
+### Prerequisite
+
+For in-tree RBD migration to CSI driver to be supported for your Kubernetes
+cluster, the Kubernetes version running in your cluster should be >= v1.23. We
+also need sidecar controllers (`csi-provisioner` and `csi-resizer`) which are
+compatible with the Kubernetes version v1.23 to be available in your cluster.
+You can enable the migration with a couple of feature gates in your Kubernetes
+cluster. These feature gates are alpha in Kubernetes 1.23 release.
+
+- `CSIMigrationRBD`: when enabled, it will redirect traffic from in-tree rbd
+  plugin (`kubernetes.io/rbd`)  to CSI driver (`rbd.csi.ceph.com`) , default
+  to `false` now.
+- `IntreePluginRBDUnregister`: Disables the RBD in-tree driver
+
+To enable feature gates,
+refer [feature gates](https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/)
+
+As a Kubernetes cluster operator that administers storage, here are the
+prerequisites that you must complete before you attempt migration to the RBD CSI
+driver:
+
+- You must install the Ceph CSI driver (`rbd.csi.ceph.com`), v3.5.0 or above,
+  into your Kubernetes cluster.
+- Considering the `clusterID` field is a required parameter for CSI driver for
+  its operations, but in-tree StorageClass has `monitors` field as a required
+  parameter, a Kubernetes storage admin has to create a clusterID based on the
+  monitors hash ( ex:`#echo -n '<monitors_string>' | md5sum`) in the CSI config
+  map and keep the monitors under this clusterID configuration. -Also, if the
+  value of `adminId` in the in-tree Storageclass is different from value `admin`
+  , the `adminSecretName` mentioned in the in-tree Storageclass has to be
+  patched with the base64 value of the `adminId` parameter value, otherwise this
+  step can be skipped.
+
+In below examples, `fast-rbd` is in-tree storageclass with provisioner
+referencing in-tree provisioner `Kubernetes.io/rbd`.
+
+```console
+$ kubectl describe sc fast-rbd |grep -i provisioner
+Provisioner:           Kubernetes.io/rbd
+```
+
+### Volume operations after enabling CSI migration
+
+This section covers the operations on volume provisioned after enabling CSI
+migration in a cluster.
+
+#### Create Volume
+
+``` console
+$ kubectl create -f pvc.yaml
+persistentvolumeclaim/testpvc created
+
+$ kubectl get pvc,pv
+NAME                               STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS   AGE
+persistentvolumeclaim/testpvc   Bound    pvc-c4e7dca5-4be6-4168-8eb5-af6ade04261f   1Gi        RWO            fast-rbd       24s
```

Comment:
looks like the indentation is off here needs to be corrected.

Reply:
the indentation still looks off please have a look at https://github.com/ceph/ceph-csi/blob/fefdb3dcc4ff59dc1030fa32f3f2fa9b77f5a2a6/docs/intree-migrate.md#volume-operations-after-enabling-csi-migration